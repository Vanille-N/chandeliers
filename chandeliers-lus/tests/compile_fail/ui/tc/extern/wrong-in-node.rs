//! Node exists but has a different signature.
use chandeliers_sem::*;

#[allow(non_camel_case_types)]
#[derive(Debug, Default)]
struct wrong_step_sig {}
impl wrong_step_sig {
    fn step(&mut self, _: ty!(i64)) -> ty!(f64) {
        unimplemented!()
    }
}
chandeliers_lus::decl! {
    extern node wrong_step_sig() returns (f: float);
}

fn main() {}
