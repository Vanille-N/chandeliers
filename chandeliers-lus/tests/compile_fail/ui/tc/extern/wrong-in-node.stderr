error[E0308]: mismatched types
   --> tests/compile_fail/ui/tc/extern/wrong-in-node.rs:14:32
    |
14  |     extern node wrong_step_sig() returns (f: float);
    |                 -------------- ^ expected `Nillable<i64>`, found `Nillable<()>`
    |                 |
    |                 arguments to this method are incorrect
    |
    = note: expected enum `Nillable<i64>`
               found enum `Nillable<()>`
note: method defined here
   --> /home/nash/School/Mpri/projects/chandeliers/chandeliers-sem/src/stepping.rs:152:8
    |
152 |     fn step(&mut self, input: <Self::Input as Embed>::Target) -> <Self::Output as Embed>::Target;
    |        ^^^^
